### PR TITLE
Rework vacation

### PR DESCRIPTION
**Pull request by chibenwa**
Current specification lacks details : 
- it is exposed to infinite loops
- it should not responds to mailing lists
- it might be a good idea to comply with personnal responders

This set of modification is inspired from RFC-5230 (Sieve vacation extension)

I also would like users to be able to specify a subject.

### CODE REVIEW

File: spec/vacationresponse.mdwn [hunk 1]
```diff
@@ -69,3 +69,7 @@ The following errors may be returned instead of the *vacationResponseSet* respon
 `accountNoMail`: Returned if the *accountId* given corresponds to a valid account, but does not contain any mail data.
 
 `invalidArguments`: Returned if one of the arguments is of the wrong type, or otherwise invalid (including using an id other than `"singleton"`). A `description` property MAY be present on the response object to help debug with an explanation of what the problem was.
+
+### Special care
+
+The implementers might avoid sending vacation notifications to mailing lists. Thus, implementations are encouraged to not send notifications to well-known addresses like "MAILER-DAEMON", "LISTSERV", "majordomo", and other addresses typically used only by automated systems.  Additionally, addresses ending in "-request" or beginning in "owner-", i.e., reserved for mailing list software, are also suggested. Implementations SHOULD NOT respond to any message that contains a "List-Id" [RFC-2919](https://www.ietf.org/rfc/rfc2919.txt), "List-Help", "List-Subscribe", "List-Unsubscribe", "List-Post", "List-Owner", or "List-Archive" [RFC-2369](https://www.ietf.org/rfc/rfc2369.txt) header field.
```
Comment (rouazana) [inline spec/vacationresponse.mdwn]: beginning with

File: spec/vacationresponse.mdwn [hunk 2]
```diff
@@ -75,3 +75,13 @@ The following errors may be returned instead of the *vacationResponseSet* respon
...
 
 Also, be careful about infinite loops. Vacation notifications should not be sent in response to vacation responses. To avoid doing so, as specified in [RFC-5230](https://tools.ietf.org/html/rfc5230), an Auto-Submitted field with a value of "auto-replied" should be included in the message header of any vacation message sent. Implementations should not send notification to e-mail with an Auto-Submitted header with a value of "auto-replied".
+
+Finaly, implementations are encouraged to comply with [RFC-3824](https://tools.ietf.org/html/rfc3834), which defines a personal responder. To do so :
```
Comment (rouazana) [inline spec/vacationresponse.mdwn]: No space before :

File: spec/vacationresponse.mdwn [hunk 3]
```diff
@@ -75,3 +75,13 @@ The following errors may be returned instead of the *vacationResponseSet* respon
...
+
+ - The Date field should be set to the date and time when the vacation response was generated.  Note that this may not be the same as the time the message was delivered to the user. 
+ - The From field should be set to the address of the owner of the Sieve script.
+ - The To field should be set to the address of the recipient of the response.
+ - An Auto-Submitted field with a value of "auto-replied" should be included in the message header of any vacation message sent.
+ - Replies MUST have the In-Reply-To field set to the Message-ID of the original message, and the References field SHOULD be updated with the Message-ID of the original message.
+
+If the original message lacks a Message-ID, an In-Reply-To need not be generated, and References need not be changed. [RFC-2822 section 3.6.4](https://tools.ietf.org/html/rfc2822#section-3.6.4) provides a complete description of how References fields should be generated.
```
Comment (rouazana) [inline spec/vacationresponse.mdwn]: need _to_ be changed?

File: spec/vacationresponse.mdwn [hunk 4]
```diff
@@ -12,6 +12,8 @@ related settings for an account. It has the following properties:
   If *isEnabled* is `true`, the date/time after which messages that arrive should receive the user's vacation response, in UTC. If `null`, the vacation response is effective immediately.
 - **toDate**: `Date|null`
   If *isEnabled* is `true`, the date/time after which messages that arrive should no longer receive the user's vacation response, in UTC. If `null`, the vacation response is effective indefinitely.
+- **subject**: `String|null`
+  The subject that will be used by the mail sent in response to messages when the vacation response is enabled. If null, an arbitrary subject can be set by the server.
```
Comment (rouazana) [inline spec/vacationresponse.mdwn:16]: The subject is not Re: [original subject without Re/Fwd/etc.]?

Comment (chibenwa) [inline spec/vacationresponse.mdwn:16]: Yes, it is by default.

But what if I want to use an other one ?

File: spec/vacationresponse.mdwn (hunk 1 above)
Comment (chibenwa) [inline spec/vacationresponse.mdwn]: Ok

File: spec/vacationresponse.mdwn (hunk 2 above)
Comment (chibenwa) [inline spec/vacationresponse.mdwn]: OK

File: spec/vacationresponse.mdwn (hunk 3 above)
Comment (chibenwa) [inline spec/vacationresponse.mdwn]: Taken directly from the Sieve Vacation RFC. I guess this is grammatically correct

File: spec/vacationresponse.mdwn (hunk 4 above)
Comment (rouazana) [inline spec/vacationresponse.mdwn:16]: ok, fine

File: spec/vacationresponse.mdwn [hunk 5]
```diff
@@ -69,3 +71,19 @@ The following errors may be returned instead of the *vacationResponseSet* respon
 `accountNoMail`: Returned if the *accountId* given corresponds to a valid account, but does not contain any mail data.
 
 `invalidArguments`: Returned if one of the arguments is of the wrong type, or otherwise invalid (including using an id other than `"singleton"`). A `description` property MAY be present on the response object to help debug with an explanation of what the problem was.
+
+### Special care
```
Comment (mbaechler) [inline spec/vacationresponse.mdwn]: I guess it belongs to "Implementing JMAP" documentation of jmap.io, don't you think ?

Comment (chibenwa) [inline spec/vacationresponse.mdwn]: ok

**Pull request closed without merging**